### PR TITLE
E2C: Start connecting on-prem to real apis

### PR DESCRIPTION
**Pull request by joshhunt**
Connects the on-prem side to the real APIs

image

What's done:
 - Call `createMigration` for user to save the auth token
 - Call `getMigrationList` to check if an auth token has been configured. Only the latest is used
 - Call `getCloudMigrationRunList` / `getCloudMigrationRun` to check if a migration has actually ran yet
    - there's a lot of futzing this data around to get it into a presentable state 😬 
    - We make an API call to `getDashboardByUID` when we render each dashboard to get the title and folder
 - Call `runCloudMigration` when the user clicks the [Upload everything] button

### CODE REVIEW

File: public/app/features/migrate-to-cloud/onprem/Page.tsx [hunk 1]
```diff
@@ -1,43 +1,172 @@
...
+
+  const handleDisconnect = useCallback(() => {
+    window.alert('TODO: Disconnect');
+  }, []);
+
+  const handleStartMigration = useCallback(() => {
+    if (migrationDestination.data?.id) {
+      actuallyRunMigration({ id: migrationDestination.data?.id });
+    } else {
+      window.alert('id still aint there');
```
Comment (ashharrison90) [inline public/app/features/migrate-to-cloud/onprem/Page.tsx:143]: how is it possible to reach this branch? 🤔

**Review by ashharrison90 — COMMENTED** [review]
think there should be some translations removed as well, probs need to run `yarn i18n:extract` 👍

File: public/app/features/migrate-to-cloud/onprem/Page.tsx (hunk 1 above)
Comment (joshhunt) [reply public/app/features/migrate-to-cloud/onprem/Page.tsx:143]: it shouldn't be, strictly. At the moment the codegen makes types with all properties optional (something to look into)

**Review by ashharrison90 — APPROVED** [review]
i'll stick approve on now as there's nothing that should block this going in 👍

**Pull request merged**